onnxruntime/core/optimizer/layer_norm_fusion.cc

```diff
@@ -345,6 +345,18 @@ X --> Pow --> ReduceMean --> Add --> Sqrt --> Div --> Mul
 |                                              ^
 |                                              |
 +----------------------------------------------+
+Additional FP16 patterns supported if allow_precision_change_ is true:
+X --> Cast1 --> Pow --> ReduceMean --> Add --> Sqrt --> Div --> Cast2 --> Mul
+        |                                               ^                  ^
+        |                                               |                  |
+        |-----------------------------------------------+                Scale
+In this pattern, we change the Mul to compute in the same type as Cast1 instead of Cast2,
+and are able to fuse the graph. We might need to add a Cast to the Scale input 
+of Mul to match the type of Cast1. We add the Cast2 after the fused Layer Norm node.
+This results in the graph:
+X ------> Cast1 --> SimplifiedLayerNormalization --> Cast2
```

Comment:
>Cast2 [](http://example.com/codeflow?start=53&length=5)

rename this as it's confusing - since Cast2 is actually removed, and new Cast node is created for LN